public/sw.js

```diff
@@ -115,12 +115,14 @@ toolbox.router.default = (request, values, options) => {
 
 // Delete old caches and claim clients so that the very first page load is controlled by a service
 // worker. (Important for responding correctly in offline state.)
-self.addEventListener('activate', () =>
-  caches.keys().then(cacheNames =>
-    Promise.all(
-      cacheNames.filter(cacheName => cacheName !== CACHE_NAME)
-        .map(cacheName => caches.delete(cacheName))
-    ).then(self.clients.claim())
+self.addEventListener('activate', event =>
+  event.waitUntil(
```

Comment:
I don't think we need to `waitUntil()` the previous caches are cleared—that should be able to happen async. What looks more suspicious to me is that `clients.claim()` does not happen immediately. I think we should be calling `clients.claim()` when we enter the activate event—the cache should be in a good state at that point.